lib/charms/identity_platform_login_ui/v0/login_ui_endpoint.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Interface library for sharing Identity Platform Login UI application's endpoint with other charms.
+This library provides a Python API for both requesting and providing a public endpoint.
+## Getting Started
+To get started using the library, you need to fetch the library using `charmcraft`.
+```shell
+cd some-charm
+charmcraft fetch-lib charms.identity_platform_login_ui.v0.login_ui_endpoint
+```
+To use the library from the requirer side:
+In the `metadata.yaml` of the charm, add the following:
+```yaml
+requires:
+  ui-endpoint-info:
+    interface: login_ui_endpoint
+    limit: 1
+```
+Then, to initialise the library:
+```python
+from charms.identity_platform_login_ui.v0.login_ui_endpoint import (
+    LoginUIEndpointRelationError,
+    LoginUIEndpointRequirer,
+)
+Class SomeCharm(CharmBase):
+    def __init__(self, *args):
+        self.login_ui_endpoint_relation = LoginUIEndpointRequirer(self)
+        self.framework.observe(self.on.some_event_emitted, self.some_event_function)
+    def some_event_function():
+        # fetch the relation info
+        try:
+            login_ui_endpoint = self.login_ui_endpoint_relation.get_login_ui_endpoint()
+        except LoginUIEndpointRelationError as error:
+            ...
+```
+"""
+
+import logging
+from typing import Dict, Optional
+
+from ops.charm import CharmBase, RelationCreatedEvent
+from ops.framework import EventBase, EventSource, Object, ObjectEvents
+from ops.model import Application
+
+# The unique Charmhub library identifier, never change it
+LIBID = "placeholder_ID"
```

Comment:
Please add the `LIBID` you registered in [this PR](https://github.com/canonical/identity-platform-login-ui-operator/pull/16)